src/main/java/com/myFirstProject/myFirstProject/service/AccountServiceImpl.java

```diff
@@ -171,20 +172,21 @@ private BigDecimal convertSum(List<CurrencyRate> currencyRates, CurrencyEntity c
     public void payForArticles(BigDecimal sum, Long id, PromoCode promoCode) {
         Account account = accountRepository.findAccountByUserId(id);
         validateAccount(account, sum);
-        BigDecimal taxPlusDiscount = findSumOfOrder(promoCode, sum);
+        promoCode.setTotalDiscount(getDiscount(promoCode, sum));
+        BigDecimal taxPlusDiscount = sum.subtract(promoCode.getTotalDiscount());
         BigDecimal subtract = account.getSum().subtract(taxPlusDiscount);
         account.setSum(subtract);
         logger.info(String.format("User with id - %d payed - %s, his account - %s", id, sum, subtract));
     }
 
-    private BigDecimal findSumOfOrder(PromoCode promoCode, BigDecimal sum) {
+    private BigDecimal getDiscount (PromoCode promoCode, BigDecimal sum) {
         if (promoCode.getPromoType() == PromoType.PERCENT) {
             BigDecimal percent = promoCode.getValue().divide(BigDecimal.valueOf(100));
-            BigDecimal fraction = BigDecimal.ONE.subtract(percent);
-            return sum.multiply(fraction);
-        } else {
+            BigDecimal discount = sum.multiply(percent);
```

Comment:
BigDecimal percent = promoCode.getValue().divide(BigDecimal.valueOf(100))  - for this line:

it make sense to keep right value in the DB instead of divide each time